server/ecore-glsp/src/main/java/com/eclipsesource/glsp/ecore/operationhandler/EcoreLabelEditOperationHandler.java

```diff
@@ -115,18 +115,18 @@ public void execute(AbstractOperationAction action, GraphicalModelState graphica
 					break;
 
 				case Types.LABEL_EDGE_NAME:
-					GEdge edge = getOrThrow(index.findElementByClass(editLabelAction.getLabelId(), GEdge.class),
-						"No edge for label with id " + editLabelAction.getLabelId() + " found");
-					EReference reference_semantic = (EReference) getOrThrow(index.getSemantic(edge),
-						"No semantic element for labelContainer with id " + edge.getId() + " found");
+					String edgeId = editLabelAction.getLabelId().split("_")[0];
```

Comment:
If the labelID is created from multiple segments, I would recommend moving the split to a helper method (e.g. ReferenceUtil.getEdgeID(labelId)). It makes it easier to understand the code, and easier to modify